ring/ring.go

```diff
@@ -286,9 +287,16 @@ func (r *Ring) starting(ctx context.Context) error {
 
 	// Start metrics update ticker, and give it a function to update the ring metrics.
 	r.metricsUpdateTicker = time.NewTicker(10 * time.Second)
```

Comment:
In the past I've seen the ticker created in a scope that allows it to be stopped with a `defer` block. That would avoid the need to store the ticker in this struct and avoid needing to stop it in the `.stopping()` method: `close(r.metricsUpdateCloser)` would be enough.

```go
	go func() {
	        ticker := time.NewTimer(10 * time.Second)
	        defer ticker.stop()

		for {
			select {
			case <-ticker.C:
				r.updateRingMetrics()
			case <-r.metricsUpdateCloser:
				return
			}
		}
	}()
```